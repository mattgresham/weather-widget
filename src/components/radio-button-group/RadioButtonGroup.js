// @flow
import React from 'react';
import RadioButton from './RadioButton';
import styles from './RadioButtonGroup.module.css';

type Values = {
  [string]: string
};

type Props = {
  onChange: Function,
  value: string,
  values: Values
};

const RadioButtonGroup = ({ onChange, value, values }: Props) => (
  <div className={styles.group}>
    {Object.keys(values).map(name => (
      <RadioButton
        checked={value === name}
        key={name}
        onChange={onChange}
        value={name}
        label={values[name]}
      />
    ))}
  </div>
);

export default RadioButtonGroup;
