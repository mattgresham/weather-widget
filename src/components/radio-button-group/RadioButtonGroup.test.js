import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import RadioButtonGroup from './RadioButtonGroup';
import TemperatureUnitConstants from '../../constants/temperatureUnits';

const mockFn = jest.fn();

it('renders correctly', () => {
  const component = render(
    <RadioButtonGroup
      onChange={mockFn}
      value="metric"
      values={TemperatureUnitConstants}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});
