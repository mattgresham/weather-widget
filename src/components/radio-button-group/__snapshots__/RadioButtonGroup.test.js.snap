// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="group"
>
  <div
    class="container"
  >
    <label
      class="label"
      for="radio-button__metric"
    >
      <input
        checked=""
        class="element"
        id="radio-button__metric"
        type="radio"
        value="metric"
      />
      °C
    </label>
  </div>
  <div
    class="container"
  >
    <label
      class="label"
      for="radio-button__imperial"
    >
      <input
        class="element"
        id="radio-button__imperial"
        type="radio"
        value="imperial"
      />
      °F
    </label>
  </div>
</div>
`;
